modules/apps/blogs/blogs-service/src/main/java/com/liferay/blogs/service/impl/BlogsEntryLocalServiceImpl.java

```diff
@@ -1763,6 +1789,24 @@ private String _getEntryURL(BlogsEntry entry, ServiceContext serviceContext)
 		).buildString();
 	}
 
+	private long _getGlobalGroudId() {
+		DynamicQuery groupDynamicQuery = _groupLocalService.dynamicQuery();
+		Property companyIdProperty = PropertyFactoryUtil.forName("companyId");
+		Property classPKProperty = PropertyFactoryUtil.forName("classPK");
+
+		groupDynamicQuery.add(companyIdProperty.eqProperty(classPKProperty));
+
+		List<Group> groups = _groupLocalService.dynamicQuery(groupDynamicQuery);
+
+		if (groups.isEmpty()) {
+			return 0;
+		}
+
+		Group globalGroup = groups.get(0);
+
+		return globalGroup.getGroupId();
+	}
+
```

Comment:
Isn't this the same thing as `company.getGroupId()`? Also, this will return the group for the company that just happens to be the first result. This is not correct; you should check the global group for the current company.